sky/provision/runpod/utils.py

```diff
@@ -0,0 +1,183 @@
+"""RunPod library wrapper for SkyPilot."""
+
+import json
+import os
+from pathlib import Path
+import time
+from typing import Dict, Optional
+
+from sky import sky_logging
+from sky.adaptors import runpod
+from sky.skylet import constants
+from sky.utils import common_utils
+
+logger = sky_logging.init_logger(__name__)
+
+GPU_NAME_MAP = {
+    'A100-80GB': 'NVIDIA A100 80GB PCIe',
+    'A100-40GB': 'NVIDIA A100-PCIE-40GB',
+    'A100-80GB-SXM': 'NVIDIA A100-SXM4-80GB',
+    'A30': 'NVIDIA A30',
+    'A40': 'NVIDIA A40',
+    'RTX3070': 'NVIDIA GeForce RTX 3070',
+    'RTX3080': 'NVIDIA GeForce RTX 3080',
+    'RTX3080Ti': 'NVIDIA GeForce RTX 3080 Ti',
+    'RTX3090': 'NVIDIA GeForce RTX 3090',
+    'RTX3090Ti': 'NVIDIA GeForce RTX 3090 Ti',
+    'RTX4070Ti': 'NVIDIA GeForce RTX 4070 Ti',
+    'RTX4080': 'NVIDIA GeForce RTX 4080',
+    'RTX4090': 'NVIDIA GeForce RTX 4090',
+    # Following instance is displayed as SXM at the console
+    # but the ID from the API appears as HBM
+    'H100-SXM': 'NVIDIA H100 80GB HBM3',
+    'H100': 'NVIDIA H100 PCIe',
+    'L4': 'NVIDIA L4',
+    'L40': 'NVIDIA L40',
+    'RTX4000-Ada-SFF': 'NVIDIA RTX 4000 SFF Ada Generation',
+    'RTX4000-Ada': 'NVIDIA RTX 4000 Ada Generation',
+    'RTX6000-Ada': 'NVIDIA RTX 6000 Ada Generation',
+    'RTXA4000': 'NVIDIA RTX A4000',
+    'RTXA4500': 'NVIDIA RTX A4500',
+    'RTXA5000': 'NVIDIA RTX A5000',
+    'RTXA6000': 'NVIDIA RTX A6000',
+    'RTX5000': 'Quadro RTX 5000',
+    'V100-16GB-FHHL': 'Tesla V100-FHHL-16GB',
+    'V100-16GB-SXM2': 'V100-SXM2-16GB',
+    'RTXA2000': 'NVIDIA RTX A2000',
+    'V100-16GB-PCIe': 'Tesla V100-PCIE-16GB'
+}
+
+
+def retry(func):
+    """Decorator to retry a function."""
+
+    def wrapper(*args, **kwargs):
+        """Wrapper for retrying a function."""
+        cnt = 0
+        while True:
+            try:
+                return func(*args, **kwargs)
+            except runpod.runpod().error.QueryError as e:
+                if cnt >= 3:
+                    raise
+                logger.warning('Retrying for exception: '
+                               f'{common_utils.format_exception(e)}.')
+                time.sleep(1)
+
+    return wrapper
+
+
+def get_set_tags(instance_id: str, new_tags: Optional[Dict]) -> Dict:
```

Comment:
```suggestion
def get_set_tags(instance_id: str, new_tags: Optional[Dict]) -> Dict[str, Any]:
```
not sure if the return type is right; pls double-check

Reply:
Good call! Actually, we don't really need the tags related functions. Just removed them. Thanks!